workflows/autometa.nf

```diff
@@ -0,0 +1,266 @@
+    *  OPTIONAL: Run Diamond BLASTp and split contigs into taxonomic groups
+    * -------------------------------------------------
+    */
+
+    if (params.taxonomy_aware) {
+        TAXON_ASSIGNMENT (
+            fasta_ch,
+            orfs_ch
+        )
+        TAXON_ASSIGNMENT.out.taxonomy
+            .set{taxonomy_results}
+        if (params.kingdom.equals('bacteria')) {
+            kmers_input_ch = TAXON_ASSIGNMENT.out.bacteria
+        } else {
+            // params.kingdom.equals('archaea')
+            kmers_input_ch = TAXON_ASSIGNMENT.out.archaea
+        }
+
+    } else {
+        kmers_input_ch = fasta_ch
+        taxonomy_results = file( "$baseDir/assets/dummy_file.txt", checkIfExists: true )
+        taxonomy_results = Channel.fromPath( taxonomy_results )
+    }
+
+    /*
+    * -------------------------------------------------
+    * Calculate k-mer frequencies
+    * -------------------------------------------------
+    */
+
+    KMERS(
+        kmers_input_ch
+    )
+    KMERS.out.normalized
+        .set{kmers_normalized_ch}
+
+    KMERS.out.embedded
+        .set{kmers_embedded_ch}
+
+
+    // --------------------------------------------------------------------------------
+    // Run hmmscan and look for marker genes in contig orfs
+    // --------------------------------------------------------------------------------
+
+    MARKERS(
+        orfs_ch
+    )
+    MARKERS.out.markers_tsv
+        .set{markers_ch}
+
+    // Prepare inputs for binning channel
+    kmers_embedded_ch
+        .join(coverage_ch)
+        .join(SEQKIT_FILTER.out.gc_content)
+        .join(markers_ch)
+        .set{binning_ch}
+    if (params.taxonomy_aware) {
+        binning_ch
+            .join(taxonomy_results)
+            .set{binning_ch}
+    } else {
+        binning_ch
+            .combine(taxonomy_results)
+            .set{binning_ch}
+    }
+
+    BINNING(
+        binning_ch
+    )
+
+    if (params.unclustered_recruitment) {
+        // Prepare inputs for recruitment channel
+        kmers_normalized_ch
+            .join(coverage_ch)
+            .join(BINNING.out.main)
+            .join(markers_ch)
+            .set{recruitment_ch}
+        if (params.taxonomy_aware) {
+            recruitment_ch
+                .join(taxonomy_results)
+                .set{recruitment_ch}
+        } else {
+            recruitment_ch
+                .combine(taxonomy_results)
+                .set{recruitment_ch}
+        }
+        RECRUIT(
+            recruitment_ch
+        )
+        RECRUIT.out.main
+            .set{binning_results_ch}
+        binning_col = Channel.value("recruited_cluster")
+    } else {
+        BINNING.out.main
+            .set{binning_results_ch}
+        binning_col = Channel.value("cluster")
+    }
+
+    // Set inputs for binning summary
+    binning_results_ch
+        .join(markers_ch)
+        .join(fasta_ch)
+        .set{binning_summary_ch}
+
+    ncbi = file(params.single_db_dir)
```

Comment:
`params.single_db_dir` isn't "required" but the pipeline fails if it's not set

Reply:
I've just fixed this, testing and will push shortly 👍 Thanks for the heads up 🛠️